packages/hardhat-core/src/internal/hardhat-network/provider/node.ts

```diff
@@ -191,6 +197,24 @@ export class HardhatNode extends EventEmitter {
           }
         }
       }
+
+      if (config.chains.has(forkNetworkId)) {
+        hardforkActivations = config.chains.get(forkNetworkId)!.hardforkHistory;
+      } else if (defaultHardhatNetworkParams.chains.has(forkNetworkId)) {
```

Comment:
This `else if` shouldn't be here, as the default config is handled by hardhat, not hardhat network, and already resolved here.

Reply:
Addressed in https://github.com/nomiclabs/hardhat/pull/1928/commits/852ae99c486919a431acbc11fc1ce60b17507313